tests/system/apmserver.py

```diff
@@ -372,6 +371,8 @@ def approve_docs(self, base_path, received):
         # The first tuple element exists to sort IDs before timestamps.
         def get_doc_id(doc):
             doc_type = doc['processor']['event']
+            if doc_type == 'metric':
+                return len(doc)
```

Comment:
What's the idea behind using the length of the doc as its ID, rather than timestamp? If you do need a separate case for metrics, can you please keep this in line with the code below, and return a tuple to sort them all?

Reply:
my bad, i thought that metricsets came without timestamp - it was just the timestamp was not set in the sample file. reverted this